groups_admin.rb

```diff
@@ -68,22 +72,43 @@
   def status_data
     {:current_time => Time.now.iso8601}
   end
-end
 
-def update_accept_header(extension,mime_type)
-  if request.url.match(/.#{extension}$/)
-    request.accept.unshift(mime_type)
-    request.path_info = request.path_info.gsub(/.#{extension}$/, '')
+  def update_accept_header(extension, mime_type)
+    if request.url.match(/.#{extension}$/)
```

Comment:
I don't think Regex are usually expensive in practice (though they could be).  I find this a clear way to specify that the match must be at the end of string.  I'm sure there are other ways too.